pkgs/misc/vscode-extensions/update_installed_exts.sh

```diff
@@ -2,23 +2,41 @@
 #! nix-shell -i bash -p curl jq coreutils
 set -eu -o pipefail
 
+# Helper to just fail with a message and non-zero exit code.
 function fail() {
     echo $1 >&2
     exit 1
 }
 
-function fetch() {
-    RES=$(curl --silent "$1")
+# Try to fetch the latest version number of the extension from Github. The is a
+# remarkably naive way of trying to achieve this but it should suffice for
+# most(?) packages. The 404 case is handled in a "weird" way because the repo
+# name may not necessarily match the extension name.
+function get_ver() {
+    PKG_URL="https://raw.githubusercontent.com/$1/$2/master/package.json"
 
-    if [[ $RES != 404* ]]; then
-        echo $RES
-    fi
-}
+    # Store the http_code and the content of the response as we need both.
+    RES=$(curl --silent --write-out "gh_pkg_json_status:%{http_code}" -X GET $PKG_URL)
 
-function get_ver() {
-    fetch "https://raw.githubusercontent.com/$1/$2/master/package.json" | jq -r '.version'
+    # Pull out the status using our (hopefully sufficiently unique) tag.
+    STATUS=$(echo "$RES" | tr -d '\n' | sed 's/.*gh_pkg_json_status://')
+    # Put the body to the side for a moment.
+    BODY=$(echo "$RES" | sed 's/gh_pkg_json_status.*//')
+
+    if [[ $STATUS -eq 200 ]]; then
+        # yay!
+        echo $BODY | jq -r '.version'
+    elif [[ $STATUS -eq 404 ]]; then
+        # Special case the 404 because that could just mean we need to add the
+        # 'vscode-' prefix to the repo name and try again
+        echo ""
+    else
+        # boo!
+        fail "Received $STATUS response when requesting package.json for $1.$2"
+    fi
 }
 
+# Try to fetch the SHA256 of the extension bundle via the VisualStudio Marketplace "API".
```

Comment:
Is it possible to do this with the version?

Reply:
We get some really good code reviews in here at nixpkgs. :smile: 

In the event that the extensions version can't be found it would be nice if it didn't fail and continued on to the next extension. Maybe just leave that particular expression incomplete (missing the version) and then print at the end what extension/s a version couldn't be found?

Reply:
I'm not sure about that, this is meant as just a helpful tool so I can see that just stubbing in a "unknown_version" in that field and reporting might be nice. But it doesn't feel like the 'nix way' to me. 

This might be used in an automated fashion, I'd rather it fail and let the human know than either introduce unexpected behaviour with defaulted string values, or even worse silently uninstalling packages (unlikely, but I worry).

Although I could make that behaviour optional... I'll have a go.

Reply:
Yeah absolutely. Maybe call it `--keep-going`.

Reply:
I can more reliably get the version with a single request now, I'll tidy this up a bit more!